presto-main/src/main/java/com/facebook/presto/operator/scalar/GroupingOperationFunction.java

```diff
@@ -30,70 +39,21 @@
 
     private GroupingOperationFunction() {}
 
-    /**
-     * The grouping function is used in conjunction with GROUPING SETS, ROLLUP and CUBE to
-     * indicate which columns are present in that grouping.
-     *
-     * <p>The grouping function must be invoked with arguments that exactly match the columns
-     * referenced in the corresponding GROUPING SET, ROLLUP or CUBE clause at the associated
-     * query level. Those column arguments are not evaluated and instead the function is
-     * re-written with the arguments below.
-     *
-     * <p>To compute the resulting bit set for a particular row, bits are assigned to the
-     * argument columns with the rightmost column being the most significant bit. For a
-     * given grouping, a bit is set to 0 if the corresponding column is included in the
-     * grouping and 1 otherwise. For an example, see the SQL documentation for the
-     * function.
-     *
-     * @param groupId An ordinal indicating which grouping is currently being processed.
-     *        Each grouping is assigned a unique monotonically increasing integer.
-     * @param columns The column arguments with which the function was
-     *        invoked converted to ordinals with respect to the base table column
-     *        ordering.
-     * @param groupingSetDescriptors A collection of ordinal lists where the index of
-     *        the list is the groupId and the list itself contains the ordinals of the
-     *        columns present in the grouping. For example: [[0, 2], [2], [0, 1, 2]]
-     *        means the the 0th list contains the set of columns that are present in
-     *        the 0th grouping.
-     * @return A bit set converted to decimal indicating which columns are present in
-     *         the grouping. If a column is NOT present in the grouping its corresponding
-     *         bit is set to 1 and to 0 if the column is present in the grouping.
-     */
     @ScalarFunction(value = INTEGER_GROUPING, deterministic = false)
     @SqlType(StandardTypes.INTEGER)
     public static long integerGrouping(
-            @SqlType(BIGINT) long groupId,
-            @SqlType("ListLiteral") List<Integer> columns,
-            @SqlType("ListLiteral") List<List<Integer>> groupingSetDescriptors)
+            @SqlType("array(integer)") Block array,
+            @SqlType(BIGINT) long groupId)
     {
-        return calculateGrouping(groupId, columns, groupingSetDescriptors);
+        return longElementAt(IntegerType.INTEGER, array, groupId);
```

Comment:
Indeed, the function shouldn't be necessary at all. It'd just be a special syntactic construct that the parser/analyzer/planner understands and gets planned as a regular array subscript operation.